pyiron_contrib/atomistic/atomicrex/fit_properties.py

```diff
@@ -0,0 +1,128 @@
+import xml.etree.ElementTree as ET
+
+from pyiron_base import InputList
+
+
+class ARFitProperty(InputList):
+    def __init__(
+        self,
+        prop,
+        target_value,
+        fit,
+        relax,
+        relative_weight,
+        residual_style,
+        output,
+        tolerance,
+        min_val,
+        max_val,
+    ):
+        super().__init__(table_name="Fit-Parameter")
+        self.prop = prop
+        self.target_value = target_value
+        self.fit = fit
+        self.relax = relax
+        self.relative_weight = relative_weight
+        self.tolerance = tolerance
+        self.residual_style = residual_style
+        self.min_val = min_val
+        self.max_val = max_val
+        self.final_value = None
+
+    
+    @property
+    def prop(self):
+        return self._prop
+    
+    @prop.setter
+    def prop(self, prop):
+        fittable_properties = [
+            "atomic-energy",
+            "atomic-forces",
+            "bulk-modulus",
+            "pressure",
+        ]
+        #cij_list = [f"c{i}{j}" for i, j in range(1,7) if j>=i]
+        #fittable_properties.extend(cij_list)
+        if prop in fittable_properties:
+            self._prop = prop
+        else:
+            raise ValueError(f"prop should be one of {fittable_properties}")
+        
+    @property
+    def residual_style(self):
+        return self._residual_style
+    
+    @residual_style.setter
+    def residual_style(self, residual_style):
+        res_styles = ["squared", "squared-relative", "absolute-diff"]
+        if residual_style in res_styles:
+            self._residual_style = residual_style
+        else:
+            raise ValueError(f"residual style has to be one of {res_styles}")
+    
+    def _is_scalar(self):
+        if self.prop != "atomic-forces":
+            return True
+        else:
+            return False
+    
+    def to_xml_element(self):
+        xml = ET.Element(f"{self.prop}")
```

Comment:
f-strings are python 3.6+ and we nominally support down to 3.4, so we either remove them or bump our python requirements.  Am open to both.

Reply:
I am fine with dropping `3.4` and `3.5` in particular for `pyiron_contrib`.